src/metabase/api/embed.clj

```diff
@@ -251,18 +251,50 @@
 
 ;;; -------------------------- Dashboard Fns used by both /api/embed and /api/preview_embed --------------------------
 
+(defn- remove-locked-and-disabled-param-values [dashboard embedding-params]
+  (let [params-to-remove (set (concat (for [[param status] embedding-params
+                                            :when          (not= status "enabled")]
+                                        param)
+                                      (for [{slug :slug} (:parameters dashboard)
+                                            :let         [param (keyword slug)]
+                                            :when        (not (contains? embedding-params param))]
+                                        param)))
+        params-ids-to-remove (set (for [parameter (:parameters dashboard)
+                                        :when     (contains? params-to-remove (keyword (:slug parameter)))]
+                                    (:id parameter)))
+        field-ids-to-remove (flatten
+                             (for [card (:dashcards dashboard)]
+                               (for [parameter-mappings (:parameter_mappings card)
+                                     :when              (contains? params-ids-to-remove (:parameter_id parameter-mappings))]
+                                 (-> (:target parameter-mappings)
+                                     (nth 1)
+                                     (nth 1)))))
+        remove-parameters (fn [dashcard]
+                            (update dashcard :parameter_mappings
+                                    (fn [param-mappings]
+                                      (into []
+                                       (filter (fn [{:keys [parameter_id]}]
+                                                 (not (contains? params-ids-to-remove parameter_id)))
+                                               param-mappings)))))]
+    (-> dashboard
+        (assoc :dashcards (map remove-parameters (:dashcards dashboard)))
+        (assoc :param_fields (apply (partial dissoc (:param_fields dashboard)) field-ids-to-remove))
```

Comment:
Usually when the inside of an `assoc` references the thing itself, `update` would work better.

I'd split these up too, since it's a little easier to read.

```suggestion
        (update :param_fields dissoc (:param_fields dashboard)))
        (update :param_fields dissoc field-ids-to-remove))
```